code/game/machinery/Sleeper.dm

```diff
@@ -145,12 +162,56 @@
 
 /obj/machinery/sleeper/New()
 	..()
+	RefreshParts()
 	spawn( 5 )
+		update_icon()
 		if(orient == "RIGHT")
-			icon_state = "sleeper_0-r"
+			generate_console(get_step(get_turf(src), WEST))
+		else
+			generate_console(get_step(get_turf(src), EAST))
 		return
 	return
 
+/obj/machinery/sleeper/Destroy()
+	..()
+	qdel(connected)
+
+/obj/machinery/sleeper/update_icon()
+	if(occupant)
+		if(orient == "LEFT")
+			icon_state = "sleeper_1"
+		else
+			icon_state = "sleeper_1-r"
+	else
+		if(orient == "LEFT")
+			icon_state = "sleeper_0"
+		else
+			icon_state = "sleeper_0-r"
+
+/obj/machinery/sleeper/proc/generate_console(turf/T as turf)
+	if(connected)
+		qdel(connected)

```

Comment:
Is there any reason to delete the console that is already connected.

You could just do
if(connected && !T.density)
connected.orient = orient
connected.update_icon()
return 1